### PR TITLE
Make teamId editable for testing

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Change teamId to readonly property and editable for internal usage and unit testing.

### CODE REVIEW

File: IdentityCore/src/util/ios/MSIDKeychainUtil.m [hunk 1]
```diff
@@ -22,30 +22,55 @@
...
 @implementation MSIDKeychainUtil
 
 #pragma mark - Public
 
-+ (NSString *)teamId
+- (instancetype)init
 {
-    static dispatch_once_t once;
-    static NSString *keychainTeamId = nil;
+    return [[self class] sharedInstance];
```
Comment (oldalton) [inline IdentityCore/src/util/ios/MSIDKeychainUtil.m]: just declare init and new as unavailable instead

File: IdentityCore/src/util/mac/MSIDKeychainUtil.m [hunk 2]
```diff
@@ -22,48 +22,71 @@
...
+#import "MSIDKeychainUtil+Internal.h"
 
 @implementation MSIDKeychainUtil
 
-+ (NSString *)teamId
+- (instancetype)init
 {
-    static dispatch_once_t once;
-    static NSString *keychainTeamId = nil;
+    return [[self class] sharedInstance];
```
Comment (oldalton) [inline IdentityCore/src/util/mac/MSIDKeychainUtil.m]: same here

File: IdentityCore/src/util/mac/MSIDKeychainUtil.m [hunk 3]
```diff
@@ -22,48 +22,71 @@
...
+    NSString *keychainTeamId = nil;
+    SecCodeRef selfCode = NULL;
+    SecCodeCopySelf(kSecCSDefaultFlags, &selfCode);
+    
+    if (selfCode)
+    {
+        CFDictionaryRef cfDic = NULL;
+        SecCodeCopySigningInformation(selfCode, kSecCSSigningInformation, &cfDic);
+        NSDictionary* signingDic = CFBridgingRelease(cfDic);
+        keychainTeamId = [signingDic objectForKey:(__bridge NSString*)kSecCodeInfoTeamIdentifier];
```
Comment (oldalton) [inline IdentityCore/src/util/mac/MSIDKeychainUtil.m]: should we log an error if either cfdic or keychainteamId is nil?

File: IdentityCore/src/util/ios/MSIDKeychainUtil.m (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/ios/MSIDKeychainUtil.m]: Done

File: IdentityCore/src/util/mac/MSIDKeychainUtil.m (hunk 2 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/mac/MSIDKeychainUtil.m]: Done

File: IdentityCore/src/util/mac/MSIDKeychainUtil.m (hunk 3 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/mac/MSIDKeychainUtil.m]: Done

**Pull request merged**